src/Psalm/Internal/Analyzer/Statements/Expression/CallAnalyzer.php

```diff
@@ -677,6 +678,41 @@ public static function applyAssertionsToContext(
                 $assertion_var_id = $assertion->var_id;
             } elseif (isset($context->vars_in_scope[$assertion->var_id])) {
                 $assertion_var_id = $assertion->var_id;
+            } elseif (str_contains($assertion->var_id, '->')) {
+                [$var_id, $property] = explode('->', $assertion->var_id);
+
+                $var_id = is_numeric($var_id) ? (int) $var_id : $var_id;
+
+                if (!is_int($var_id) || !isset($args[$var_id])) {
+                    continue; // any chance to report malformed/not supported assertion?
```

Comment:
You probably could push another issue into `IssueBuffer`, e.g. `InvalidDocblock`. If you don't have access to precise assertion location, method/function location could be used.

Reply:
Whooohoo!
```InvalidDocblock - src/somefile.php:18:21 - Property bx is not defined on variable 0 so assertion cannot be applied```

Thanks for the advice!